packages/core/src/App/Containers/RealAccountSignup/helpers/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Takes validation result of a form, sets warnings on component's state and pass errors down to the form
+ *
+ * @param {object} values - object containing form field values and validations
+ * @return {object} errors
+ */
+function setWarnsFilterErrors(values) {
+    const warnings = {};
+    Object.keys(values).forEach(field => {
+        if (values[field][0] === 'warn') {
+            warnings[field] = values[field][1];
+            delete values[field];
+        }
+    });
+    this.setState({ warnings });
```

Comment:
Setting state here does not look good, as there might be a consumer to this function that doesn't rely on state and just wants to filter out warnings from errors. Please consider using a pure function for this. 

We can defer the responsibility of storing the data to the consumer. something like: 

```js
values =>  splitErrorTypes(this.props.validate(values)).errors
```
or, 
```js
values => this.storeWarningsInTheStateAndReturnErrors(splitErrorTypes(this.props.validate(values))) // pardon the function naming
```